File: app/src/main/java/com/ashwinrao/packup/data/repo/ItemRepository.java
package com.ashwinrao.packup.data.repo;

import androidx.lifecycle.LiveData;

import com.ashwinrao.packup.data.AppDatabase;
import com.ashwinrao.packup.data.AsyncTasks;
import com.ashwinrao.packup.data.model.Item;
import com.ashwinrao.packup.data.repo.dao.ItemDao;

import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class ItemRepository {

    private ItemDao dao;

    @Inject
    public ItemRepository(AppDatabase database) {
        dao = database.itemDao();
    }

    public LiveData<List<Item>> getItems() {
        return dao.getItems();
    }

    public LiveData<Item> getItem(String name) {
        return dao.getItem(name);
    }

    public LiveData<List<Item>> getItemsValued(double estimatedValue) {
        return dao.getItemsValued(estimatedValue);
    }

    public LiveData<List<Item>> getItemsFromBox(String boxUUID) {
        return dao.getItemsFromBox(boxUUID);
    }

    public void insert(Item... items) {
        new AsyncTasks.ItemAsyncTask(dao, "insert").execute(items);
    }

    public void update(Item... items) {
        new AsyncTasks.ItemAsyncTask(dao, "update").execute(items);
    }

    public void delete(Item... items) {
        new AsyncTasks.ItemAsyncTask(dao, "delete").execute(items);
    }




}

File: app/src/main/java/com/ashwinrao/packup/data/repo/ItemRepository.kt
package com.ashwinrao.packup.data.repo

import androidx.lifecycle.LiveData
import com.ashwinrao.packup.data.AppDatabase
import com.ashwinrao.packup.data.model.Item
import com.ashwinrao.packup.data.repo.dao.ItemDao
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ItemRepository @Inject constructor(database: AppDatabase) {

    private val dao: ItemDao = database.itemDao()

    fun getItems(): LiveData<List<Item>> = dao.getItems()

    fun getItemsFromBox(boxUUID: String) : LiveData<List<Item>> =
            dao.getItemsFromBox(boxUUID)

    fun insert(vararg item: Item) =
            CoroutineScope(Dispatchers.IO)
                    .launch { dao.insert(*item) }

    fun update(vararg item: Item) =
            CoroutineScope(Dispatchers.IO)
                    .launch { dao.update(*item) }

    fun delete(vararg item: Item) =
            CoroutineScope(Dispatchers.IO)
                    .launch { dao.delete(*item) }
}
